src/app/shared/components/create-project/create-project.component.ts

```diff
@@ -17,90 +15,102 @@ export class CreateProjectComponent implements OnInit {
   @Input() isVisible!: boolean;
   @Output() onToggle = new EventEmitter<boolean>();
 
-  primarySkills: PrimarySkill[] = [];
-  isPrimarySkillsValid: boolean = false;
+  primarySkills: Map<string, FormGroup> = new Map();
   isPrimarySkillsTouched: boolean = false;
-  data!: CreateProject;
+  data: CreateProject | undefined;
   form!: FormGroup;
 
-  primarySkillsValidity$ = new Subject<boolean>();
-
   constructor(
     private fb: FormBuilder,
     private projectsService: ProjectsService,
     private message: NzMessageService
   ) {}
 
-  private checkPrimarySkillsValidity(): boolean {
-    return this.primarySkills.every((skill) => {
-      return skill.name && skill.description && skill.link;
-    });
-  }
-
-  onPrimarySkillAdd() {
+  onPrimarySkillToggle(value: boolean, primarySkill: PrimarySkill) {
```

Comment:
value - this name is good to use in loops, arrow function but not for methods. Please give more specific names. e.g. isPrivarySkill, isOpen ...